### PR TITLE
Augment Docker error handling

### LINKED ISSUE #47: Update Docker error handling
The Docker module should be updated to include better error handling of various situations a user could run into while running the application. The following is a non-comprehensive list of additional scenarios that should be handled instead of throwing the ugly Docker errors:

1. The Docker daemon is not running.
2. The Bobber container is not running.
3. Issues adding NVIDIA resources to the container.
4. Version mismatch between the Bobber application and the running container.

### PR DESCRIPTION
**Pull request by roclark**
The Docker module needs extra error handling to help point users in the right direction when common errors pop up, like missing containers, version mismatches, and communication errors.

Additionally, the exit codes need to be updated to positive numbers in the range of 0-127 to be properly enumerated by the system. See the [Python docs](https://docs.python.org/3/library/sys.html#sys.exit) for more info.

Closes #47

Signed-Off-By: Robert Clark <roclark@nvidia.com>

### CODE REVIEW

**Comment by roclark** [comment]
Now properly handling the following scenarios:

1. The Docker daemon is not running:
```
$ bobber run-nccl test localhost
Error: Could not communicate with the Docker daemon.
Ensure Docker is running with "systemctl start docker"
```
2. The Bobber container is not running:
```
$ bobber run-nccl test localhost
Bobber container not running. Launch a container with "bobber cast" prior to running any tests.
```
3. The NVIDIA runtime could not be added to the container:
```
$ bobber cast /raid
NVIDIA container runtime not found. Ensure the latest nvidia-docker libraries and NVIDIA drivers are installed.
```
4. The Bobber container and application have mismatched versions:
```
$ bobber run-nccl test localhost
Bobber container version mismatch.
Kill the running Bobber container with "docker kill bobber" and re-cast a new container with "bobber cast" prior to running any tests.
```

File: bobber/lib/exit_codes.py [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # SPDX-License-Identifier: MIT
 # This file contains a list of exit codes for debugging
 SUCCESS = 0  # Successful termination
-BASELINE_FAILURE = -10  # Performance did not meet criteria
-MISSING_LOG_FILES = -20  # Parsing directory with no logs
-DOCKER_BUILD_FAILURE = -30  # Failure building Docker image
+BASELINE_FAILURE = 10  # Performance did not meet criteria
```
Comment (joehandzik) [inline bobber/lib/exit_codes.py:4]: So we switched back to positive numbers here? Any particular reason?

Comment (roclark) [reply bobber/lib/exit_codes.py:4]: Correct. I put the explanation in the commit message and the PR, but basically Python doesn't handle negative exit codes and throws a semi-random number instead, so by switching back to positive numbers we get the actual exit values we expect.

Comment (joehandzik) [reply bobber/lib/exit_codes.py:4]: My mistake - thanks for the clarity!

**Review by joehandzik — APPROVED** [review]

File: bobber/lib/exit_codes.py (hunk 1 above)
Comment (roclark) [reply bobber/lib/exit_codes.py:4]: No worries! Great catch!

**Pull request merged**